src/domain/wallet-on-chain-addresses/wallet-address-receiver.ts

```diff
@@ -0,0 +1,62 @@
+import { PriceRatio } from "@domain/payments"
+import { AmountCalculator, WalletCurrency } from "@domain/shared"
+
+const calc = AmountCalculator()
+
+export const WalletAddressReceiver = async <S extends WalletCurrency>({
+  walletAddress,
+  receivedBtc,
+  feeBtc,
+  usdFromBtc,
+  usdFromBtcMidPrice,
+}: WalletAddressReceiverArgs<S>): Promise<
+  WalletAddressReceiver<S> | DealerPriceServiceError | ValidationError
+> => {
+  if (walletAddress.recipientWalletDescriptor.currency === WalletCurrency.Btc) {
+    const receivedUsd = await usdFromBtcMidPrice(receivedBtc)
+    if (receivedUsd instanceof Error) return receivedUsd
+
+    const priceRatio = PriceRatio({ usd: receivedUsd, btc: receivedBtc })
+    if (priceRatio instanceof Error) return priceRatio
+
+    const bankFee = {
+      usdBankFee: priceRatio.convertFromBtc(feeBtc),
+      btcBankFee: feeBtc,
+    }
+
+    return {
+      ...walletAddress,
+      btcToCreditReceiver: calc.sub(receivedBtc, feeBtc),
+      usdToCreditReceiver: calc.sub(receivedUsd, bankFee.usdBankFee),
+      recipientWalletDescriptor: walletAddress.recipientWalletDescriptor,
+      ...bankFee,
+      receivedAmount: () =>
```

Comment:
Why is this lazy. the `xxxDescriptor.currency` can never change during the lifecycle of this state.

Reply:
It's a pattern we seem to use in the `WalletInvoiceReceiver` and places like `PaymentFlow` to access internal values.